src/Starcounter.Linq/QueryExecutor.cs

```diff
@@ -13,7 +16,8 @@ public object Execute<TResult>(string sql, object[] variables, QueryResultMethod
                 return null;
             }
 
-            if (typeof(TResult).IsGenericType)
+            var resultType = typeof(TResult);
+            if (resultType == enumerableType || resultType.GetInterfaces().Contains(enumerableType))

```

Comment:
Should not `Type.IsAssignableFrom` be used for this check? As far as I know this check won't pass if the result implements a custom interface which implements `IEnumerable`. 

https://msdn.microsoft.com/en-us/library/system.type.isassignablefrom(v=vs.110).aspx

Reply:
Sure, thanks. Fixed it.